### PR TITLE
Fix the problem of out-of-memory and the number of workers used by image_net_trainer

### PR DESCRIPTION
**Pull request by shiannn**
Add .item() to avoid out-of-memory problem on cpu/gpu due to accumulating gradients of loss
Add num_workers parameter to ImageNetEvaluator to enable modifying the number of processes used

### CODE REVIEW

File: Examples/torch/utils/image_net_trainer.py [hunk 1]
```diff
@@ -113,7 +114,7 @@ def _train_loop(self, model: nn.Module, criterion: torch.nn.modules.loss, optimi
                 # compute model output
                 output = model(images)
                 loss = criterion(output, target)
-                avg_loss += loss
+                avg_loss += loss.item()
```
Comment (shiannn) [inline Examples/torch/utils/image_net_trainer.py:117]: add .item() to detach the computation graph accumulated by torch

File: Examples/torch/utils/image_net_trainer.py [hunk 2]
```diff
@@ -71,7 +71,8 @@ def __init__(self, images_dir: str, image_size: int, batch_size: int = 128,
                                                 is_training=True, num_workers=num_workers,
                                                 num_samples_per_class=num_train_samples_per_class).data_loader
 
-        self._evaluator = ImageNetEvaluator(images_dir=images_dir, image_size=image_size, batch_size=batch_size)
+        self._evaluator = ImageNetEvaluator(images_dir=images_dir, image_size=image_size, batch_size=batch_size, 
+                                            num_workers=num_workers)
```
Comment (shiannn) [inline Examples/torch/utils/image_net_trainer.py:75]: add param `num_workers` to prevent always using the default number of `num_workers` (32)

**Comment by shiannn** [comment]
fix #2046

**Review by quic-mangal — DISMISSED** [review]

**Review by quic-akhobare — DISMISSED** [review]

File: Examples/torch/utils/image_net_trainer.py (hunk 2 above)
Comment (quic-akhobare) [reply Examples/torch/utils/image_net_trainer.py:75]: Hi @shiannn - the pylint checker is very sensitive :) Its complaining that you have trailing whitespace on this line. Please remove that and push again. Thanks.

Comment (shiannn) [reply Examples/torch/utils/image_net_trainer.py:75]: Hi @quic-akhobare - Fixed. Thanks!

**Review by quic-akhobare — APPROVED** [review]

**Comment by quic-akhobare** [comment]
Thanks. Approved and merged for you.
